### PR TITLE
HCAP-677 | Rest API to assign participant to a cohort

### PR DESCRIPTION
**Pull request by fwpushan**
1. Add api routes
2. Add service method
3. Add tests

Ticket: [HCAP-677](https://freshworks.atlassian.net/browse/HCAP-677)

### CODE REVIEW

File: server/routes/cohorts.js [hunk 1]
```diff
@@ -57,4 +62,63 @@ router.get(
...
+// Assign participant
+router.post(
+  '/:id/assign/:participantId',
+  [applyMiddleware(keycloak.allowRolesMiddleware('health_authority', 'employer'))],
+  asyncMiddleware(async (req, res) => {
+    const { user_id: userId, sub: localUserId } = req.user;
+    const user = userId || localUserId;
+    const { id, participantId } = req.params;
+    if (id && participantId) {
+      const [participant] = await getParticipantByID({ id: +participantId });
```
Comment (Zenen-freshworks) [inline server/routes/cohorts.js:74]: what does the `+` operator do here? I see you using it on line 78 as well.

**Review by Zenen-freshworks — APPROVED** [review]
Yep, looks good. I just have one question about a `+` operator you used.

File: server/routes/cohorts.js (hunk 1 above)
Comment (fwpushan) [reply server/routes/cohorts.js:74]: convert number to string

File: server/routes/cohorts.js [hunk 2]
```diff
@@ -57,4 +62,63 @@ router.get(
...
+    if (!participant) {
+      return res.status(400).send('Invalid participant id');
+    }
+    const [cohort] = (await getAssignCohort({ participantId: participant.id })) || [{}];
+    logger.info({
+      action: 'cohort_participant_get_assigned',
+      performed_by: {
+        user,
+      },
+      cohortId: cohort.id || '',
```
Comment (vyasworks) [inline server/routes/cohorts.js:117]: is it possible that `cohort.id` could be undefined here?

**Review by vyasworks — APPROVED** [review]

File: server/routes/cohorts.js (hunk 2 above)
Comment (fwpushan) [reply server/routes/cohorts.js:117]: Yes, if no cohort assigned ....

**Pull request merged**